integration/setup/tenant_cluster.go

```diff
@@ -295,3 +338,503 @@ func ensureCertConfigsInstalled(ctx context.Context, id string, config Config) e
+
+	var subnetID string
+	var vpcID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting for public subnet and vpc")
+
+		i := &ec2.DescribeSubnetsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{aws.String(clusterID)},
+				},
+				{
+					Name:   aws.String("tag:aws:cloudformation:logical-id"),
+					Values: []*string{aws.String("PublicSubnet")},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeSubnets(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(o.Subnets) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one subnet, got %d", len(o.Subnets))
+		}
+
+		subnetID = *o.Subnets[0].SubnetId
+		vpcID = *o.Subnets[0].VpcId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for public subnet %#q and vpc %#q", subnetID, vpcID))
+	}
+
+	var workerSecurityGroupID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting for worker security group")
+
+		i := &ec2.DescribeSecurityGroupsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{aws.String(clusterID)},
+				},
+				{
+					Name:   aws.String("tag:aws:cloudformation:logical-id"),
+					Values: []*string{aws.String("WorkerSecurityGroup")},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeSecurityGroups(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(o.SecurityGroups) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one security group, got %d", len(o.SecurityGroups))
+		}
+
+		workerSecurityGroupID = *o.SecurityGroups[0].GroupId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for worker security group %#q", workerSecurityGroupID))
+	}
+
+	var bastionSecurityGroupID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating bastion security group")
+
+		i := &ec2.CreateSecurityGroupInput{
+			Description: aws.String("Allow SSH access from everywhere to port 22."),
+			GroupName:   aws.String(clusterID + "-bastion"),
+			VpcId:       aws.String(vpcID),
+		}
+
+		o, err := config.AWSClient.EC2.CreateSecurityGroup(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		bastionSecurityGroupID = *o.GroupId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created bastion security group %#q", bastionSecurityGroupID))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "tagging bastion security group")
+
+		i := &ec2.CreateTagsInput{
+			Resources: []*string{
+				aws.String(bastionSecurityGroupID),
+			},
+			Tags: []*ec2.Tag{
+				{
+					Key:   aws.String("Name"),
+					Value: aws.String(clusterID + "-bastion"),
+				},
+				{
+					Key:   aws.String("giantswarm.io/cluster"),
+					Value: aws.String(clusterID),
+				},
+			},
+		}
+
+		_, err = config.AWSClient.EC2.CreateTags(i)
```

Comment:
This part seems fragile. If this call fails we'll create an orphaned SG.  (same below). I think it will be deleted with the VPC so I'm no super worried but can't we get SG group ID from the name?

Reply:
The stupid thing with the security groups is that you cannot set the name when creating it. The name is set here with the tags. Group name and description as set above are different and cannot be used to search the security group. I agree it is bullshit but I did not find another way of doing things.

Reply:
If it wasn't testing code I would even try to list all SGs and check name property. Otherwise the code isn't idempotent. But here in short-living VPC I don't see much risk. It would be good to adda comment though in case someone wants to copy/paste that code in the future.